docs/docs/migrate-from-v1/README.md

```diff
@@ -0,0 +1,259 @@
+  this.name = 'Dylan';
+  // downstream steps could use steps.get_customer_data.name to retrieve 'Dylan'
+}
+```
+
+In v2, this specific way of exporting data is now available with `$.export`:
+
+```javascript
+// this step's name is get_customer_data
+defineComponent({
+  async run({ steps, $ }) {
+    $.export('name', 'Dylan');
+    // downstream steps can use steps.get_customer_data.name to retrieve 'Dylan'
+  }
+});
+```
+::: tip
+Using `return` to export data is the same from v1 to v2. You can still `return` data, and it will be available to other steps with `steps.[stepName].$return_value.
+:::
+
+### Exiting a workflow early 
+
+In v1, the `$exit` function can be called to exit a flow early:
+
+```javascript
+async (event, steps) {
+  exit('Exiting the whole workflow early');
+  console.log('I will never run');
+}
+```
+
+In v2, this same function is available, but under `$.exit`:
+
+```javascript
+defineComponent({
+  async run({ steps, $ }) {
+    $.flow.exit('Exiting the workflow early');
+    console.log('I will never run');
+  }
+});
+```
+
+::: warning
+In v2, `$end` does not display the reason in the workflow results logs.
+::: 
+
+### Params vs Props
+
+In workflow builder v2, both Component **props** and Node.js code steps **props** are one in the same.
+
+You can still enter free text and select data from other steps in pre-built actions. Also can add your own custom props that accept input like strings, numbers and more just like in v1.
+
+#### Defining params
+
+In the v1 workflow builder, params could be structured or unstructured. The [params schema builder](https://pipedream.com/docs/v1/workflows/steps/params/#configuring-custom-params) allowed you to add your own custom params to steps.
+
+In v2, you can add your own custom props without leaving the code editor with `props`.
+
+```javascript
+export default defineComponent({
+  props: {
+    firstName: {
+      type: 'string',
+      label: 'Your first name',
+    }
+  },
+  async run({ steps, $ }) {
+    console.log(this.firstName);
+  }
+});
+
+```
+
+In the example above a `firstName` string prop is created. The value assigned to this prop in the workflow builder by either a static string or dynamic data will be available in the step as `this.firstName`.
+
+Additionally, a visual component is rendered in the step **Configuration** to accept this input:
+
+<img src="./images/custom-string-prop.png" alt="Custom props render in the Configuration portion of the code step.">
+
+### Connecting apps
+
+In the v2 builder, you can connect apps with your code using [props](/components/props).
+
+Above the `run` function define an app prop that your Node.js step integrates with.
+
+```javascript
+export default defineComponent({
+  props: {
+    twitter: {
+      type: "app",
+      app: "slack",
+    }
+  },
+  async run({ steps, $ }) {
+    // now your Slack token for authenticating API calls will be available here:
+    this.slack.$auth.oauth_access_token;
```

Comment:
I would recommend modifying this code since I don't think it will work as written. Generally I wouldn't `console.log` it since it's technically credentials, but maybe import the Slack SDK and use the access token to instantiate a Slack client.

Reply:
Yea agreed, even though it's something I myself do, we can't say to do it publicly.

The path looks off but it's correct, I just added the exact example from our default action.